sympy/printing/codeprinter.py

```diff
@@ -502,7 +502,11 @@ def _print_Mul(self, expr):
             return sign + '*'.join(a_str) + "/(%s)" % '*'.join(b_str)
 
     def _print_not_supported(self, expr):
-        self._not_supported.add(expr)
+        try:
+            self._not_supported.add(expr)
+        except TypeError:
+            # not hashable
```

Comment:
I mostly want to remove `except TypeError` from the codebase but this is an example of correct use. TypeError is the expected exception for unhashable objects and we collect immediately around the call to `add`. Most importantly there is a comment explaining why the exception is being caught.

Is the `_not_supported` set thread-safe? It looks like it's stored as an instance attribute during the call to `self._print` and then reset afterwards. Does that mean that using the same printer instance from two threads would lead to conflicts around global state?

Also would it make more sense just to raise an error when something isn't supported?

Reply:
I haven't put much thought into it, I added this because my new test crashed without it. I think the idea is that all the unsupported objects get printed at the end, and then you can fill in the gaps by hand.